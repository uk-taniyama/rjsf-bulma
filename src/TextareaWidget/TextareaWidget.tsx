import { getUiOptions, WidgetProps } from "@rjsf/utils";
import { FieldGroup, FieldControl, FieldLabel } from "../ui";

type CustomWidgetProps = WidgetProps & {
  options: any;
};

const TextareaWidget = ({
  id,
  placeholder,
  value,
  required,
  disabled,
  autofocus,
  label,
  readonly,
  onBlur,
  onFocus,
  onChange,
  options,
  schema,
  rawErrors = [],
  uiSchema,
}: CustomWidgetProps) => {
  const _onChange = ({
    target: { value },
  }: React.ChangeEvent<HTMLTextAreaElement>) =>
    onChange(value === "" ? options.emptyValue : value);
  const _onBlur = ({
    target: { value },
  }: React.FocusEvent<HTMLTextAreaElement>) => onBlur(id, value);
  const _onFocus = ({
    target: { value },
  }: React.FocusEvent<HTMLTextAreaElement>) => onFocus(id, value);

  return (
    <FieldGroup>
      <FieldLabel id={id} label={label} schema={schema} uiSchema={uiSchema} required={required} />
      <FieldControl>
        <textarea
          id={id}
          name={id}
          placeholder={placeholder}
          disabled={disabled}
          readOnly={readonly}
          value={value}
          required={required}
          autoFocus={autofocus}
          rows={options.rows || 5}
          onChange={_onChange}
          onBlur={_onBlur}
          onFocus={_onFocus}
        />
      </FieldControl>
    </FieldGroup>
  );
};

export default TextareaWidget;
