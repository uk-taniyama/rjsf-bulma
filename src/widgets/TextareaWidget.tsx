import type { FC } from "react";
import { WidgetProps } from "@rjsf/utils";
import { FieldControl } from "../ui";

const TextareaWidget: FC<WidgetProps> = ({
  id,
  options,
  value,
  placeholder,
  autofocus,
  required,
  disabled,
  readonly,
  onChange,
  onBlur,
  onFocus,
}) => {
  const _onChange = ({
    target: { value },
  }: React.ChangeEvent<HTMLTextAreaElement>) =>
    onChange(value === "" ? options.emptyValue : value);
  const _onBlur = ({
    target: { value },
  }: React.FocusEvent<HTMLTextAreaElement>) => onBlur(id, value);
  const _onFocus = ({
    target: { value },
  }: React.FocusEvent<HTMLTextAreaElement>) => onFocus(id, value);

  return (
    <>
      <FieldControl>
        <textarea
          id={id}
          name={id}
          className="textarea is-small"
          value={value == null || value === options.emptyValue ? "" : value}
          placeholder={placeholder}
          required={required}
          disabled={disabled}
          readOnly={readonly}
          autoFocus={autofocus}
          rows={options.rows || 5}
          onChange={_onChange}
          onBlur={_onBlur}
          onFocus={_onFocus}
        />
      </FieldControl>
    </>
  );
};

export default TextareaWidget;
