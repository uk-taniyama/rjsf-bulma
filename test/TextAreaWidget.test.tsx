import React from "react";

import renderer from "react-test-renderer";

import TextareaWidget from "../src/widgets/TextareaWidget";

import { makeWidgetMockProps } from "./helpers/createMocks";

describe("TextareaWidget", () => {
  test("simple without errors", () => {
    const tree = renderer
      .create(<TextareaWidget {...makeWidgetMockProps({})} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("simple with errors", () => {
    const tree = renderer
      .create(
        <TextareaWidget
          {...makeWidgetMockProps({ rawErrors: ["Invalid 1", "Invalid 2"] })}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("simple without required", () => {
    const tree = renderer
      .create(<TextareaWidget {...makeWidgetMockProps({ required: false })} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
