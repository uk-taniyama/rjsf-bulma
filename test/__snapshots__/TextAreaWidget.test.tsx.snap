// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextareaWidget simple with errors 1`] = `
<div
  className="control"
>
  <textarea
    autoFocus={true}
    className="textarea is-danger"
    disabled={false}
    id="_id"
    name="_id"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder=""
    readOnly={true}
    required={true}
    rows={5}
    value="value"
  />
</div>
`;

exports[`TextareaWidget simple without errors 1`] = `
<div
  className="control"
>
  <textarea
    autoFocus={true}
    className="textarea is-danger"
    disabled={false}
    id="_id"
    name="_id"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder=""
    readOnly={true}
    required={true}
    rows={5}
    value="value"
  />
</div>
`;

exports[`TextareaWidget simple without required 1`] = `
<div
  className="control"
>
  <textarea
    autoFocus={true}
    className="textarea is-danger"
    disabled={false}
    id="_id"
    name="_id"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder=""
    readOnly={true}
    required={false}
    rows={5}
    value="value"
  />
</div>
`;
